cocotb/triggers.py

```diff
@@ -729,3 +729,74 @@ def _wait(self):
         for _ in range(self.num_cycles):
             yield trigger
         raise ReturnValue(self)
+
+
+class StableCondition(Waitable):
+    """
+    Wait until a given condition is True in ReadOnly state.
+
+    The trigger will first do ``yield ReadOnly()`` and then do the check.
+
+    Args:
+        condition (callable): Output of condition called without any parameters
+            is checked to be equivalent of logic True for condition to be met.
+        event (Trigger): event after which to check the condition; defaults to
+            nexttimestep.
+        extra_event (bool): wether to wait for an extra event after condition
+            is met; defaults to ``False``.
+
+    Example:
+
+    .. code-block:: python
+
+        # Wait for ack
+        yield StableCondition(
+            lambda: bus.cycle.integer and bus.ack.integer,
+            event=RisingEdge(dut.clk)
+        )
+
+    .. versionadded: 1.3
+    """
+    def __init__(self, condition, **kwargs):
+        event = kwargs.pop('event', NextTimeStep())
+        extra_event = kwargs.pop('extra_event', False)
+        reject_remaining_kwargs('StableCondition.__init__', kwargs)
+        self.condition = condition
+        self.event = event
+        self.extra_event = extra_event
+
+    @decorators.coroutine
+    def _wait(self):
+        while True:
+            yield ReadOnly()
+            if self.condition():
+                break
+            yield self.event
+        if self.extra_event:
+            yield self.event
+
+
+class StableValue(Waitable):
+    """
+    Wait until a signal has a given value in ReadOnly state.
+
+    The trigger will first do ``yield ReadOnly()`` and then do the comparison.
+
+    Args:
+        signal (SimHandle): the signal to monitor.
+        event (Trigger): event after which next is done; default is waiting for
+            a change of the given signal; defaults to ``Edge(signal)``.
+        extra_event (bool): wether to wait for an extra event after value has the
+            right value; defaults to ``False``.
+
+    .. versionadded: 1.3
+    """
+    def __init__(self, signal, value, **kwargs):
+        if value not in (0, 1):
+            raise ValueError("value has to be 0 or 1")
+        event = kwargs.pop('event', Edge(signal))
+        extra_event = kwargs.pop('extra_event', False)
+        reject_remaining_kwargs('StableCondition.__init__', kwargs)
```

Comment:
It might make more sense to just forward `**kwargs` here

Reply:
Will do, but will cause more confusing error messages when people are making this mistake.

Reply:
Brought back popping of ``event`` as it changes the default value. See discussion below.